apps/percona/chaos/node_freeze/test.yml

```diff
@@ -0,0 +1,153 @@
+          include_tasks: "/common/utils/mysql_data_persistence.yml"
+          vars:
+            status: 'LOAD'
+            ns: "{{ app_ns }}"
+            pod_name: "{{ app_pod_name.stdout }}"
+            dbuser: 'root'
+            dbpassword: 'k8sDem0'
+            dbname: "tdb{{ uniqstr.stdout }}"
+          when: data_persistance != ''
+
+        ## STORAGE FAULT INJECTION 
+
+        - include_tasks: /chaoslib/chaoskube/node_freeze_chaos.yml
+          vars:
+            nodeaction: "memory-freeze"      
+            namespace: "{{ app_ns }}"
+
+        - name: wait for app pod to reschedule
+          shell: >
+            sleep {{ f_duration | int + 100 }}
+          args:
+            executable: /bin/bash
+
+        - name: Verify AUT liveness post fault-injection
+          shell: >
+            kubectl get pods -n {{ app_ns }} -l {{ app_label }} --no-headers
+            -o custom-columns=:status.phase
+          args:
+            executable: /bin/bash
+          register: app_status
+          until: "((app_status.stdout_lines|unique)|length) == 1 and 'Running' in app_status.stdout"
+          delay: 10
+          retries: 12
+
+        - include_tasks: /common/utils/application_liveness_check.yml
+          when: liveness_label != ''
+
+        - name: Get application pod name
+          shell: >
+            kubectl get pods -n {{ app_ns }} -l {{ app_label }} --no-headers
+            -o=custom-columns=NAME:".metadata.name"
+          args:
+            executable: /bin/bash
+          register: rescheduled_app_pod
+
+        - name: Verify mysql data persistence
+          include_tasks: "/common/utils/mysql_data_persistence.yml"
+          vars:
+            status: 'VERIFY'
+            ns: "{{ app_ns }}"
+            pod_name: "{{ rescheduled_app_pod.stdout }}"
+            dbuser: 'root'
+            dbpassword: 'k8sDem0'
+            dbname: "tdb{{ uniqstr.stdout }}"
+          when: data_persistance != ''
+
+        - set_fact:
+            flag: "Pass"
+
+      rescue: 
+        - set_fact: 
+            flag: "Fail"
+
+      always:
+
+        - block:
+
+            # Get AWS instance Id of the Application node
+            - name: Get the AWS instance Id of APP NOde
+              shell: kubectl get node {{ app_node }} -o jsonpath={.spec.providerID} | cut -d '/' -f 5
+              args:
+                executable: /bin/bash
+              register: instance_id
+
+            - name: Get region from app node
+              shell: >
+                 kubectl get nodes {{ app_node }}
+                 -o jsonpath='{.metadata.labels.failure-domain\.beta\.kubernetes\.io\/region}'
+              register: region
+
+            - set_fact:
+                instance_id: "{{ instance_id.stdout }}"
+                region: "{{ region.stdout }}"
+
+            # Execute the chaos util
+
+            - include_tasks: /chaoslib/aws_chaos/chaosutil_aws.yml
+              vars:
+                action: "restarted"
```

Comment:
@vibhor995 Can we have a check to verify node status post recovery?

Reply:
Yes @dargasudarshan I am including a check to verify node status post recovery.